src/virtual-machine.js

```diff
@@ -284,6 +284,20 @@ class VirtualMachine extends EventEmitter {
     }
 
     /**
+     * Delete a costume from the current editing target.
+     * @param {int} costumeIndex - the index of the costume to be removed.
+     */
+    deleteCostume (costumeIndex) {
+        this.editingTarget.sprite.costumes = this.editingTarget.sprite.costumes
+            .slice(0, costumeIndex)
+            .concat(this.editingTarget.sprite.costumes.slice(costumeIndex + 1));
+        if (costumeIndex === this.editingTarget.currentCostume) {
+            this.editingTarget.setCostume(costumeIndex - 1);
+        }
```

Comment:
Would you please add some tests? Curious about these cases:
* `costumeIndex === currentCostume === 0`: Would expect currentCostume to remain 0, but I believe it will wrap to the last costume
* `costumeIndex === 0`, `currentCostume === 2`, `costumes.length === 3`: Would expect currentCostume to decrease to maintain the same costume, but I believe it will wrap to 0
* `costumes.length === 1`: We should probably reject attempts to delete the last costume?